Update emeritus contributors list

tatiana opened the pull request:
There have been people who had a major role in the creation and/or development of Cosmos in previous phases, but for different circumstances, they have not been able to continue contributing to it in more recent times. This is not only from a code perspective, but also from a community perspective:

<img width="410" alt="Screenshot 2024-05-14 at 11 38 12" src="https://github.com/astronomer/astronomer-cosmos/assets/272048/c8312fcb-db7d-44bf-b18a-a5867dbbbb4e">
<img width="413" alt="Screenshot 2024-05-14 at 11 38 03" src="https://github.com/astronomer/astronomer-cosmos/assets/272048/48e43f7d-9625-4139-a4b3-5c2b4b92662e">

With this change, we recognize their contributions. If, in the future, they start to contribute again to the project, we'll move them to the contributions section again, regaining committer superpowers.

More information in the contributors roles can be found at: https://github.com/astronomer/astronomer-cosmos/blob/main/docs/contributors-roles.rst

ashb commented:
I'm not so sure that Harel should have this status -- he hasn't contributed directly to the project much. My understanding was his involvement was because he was a team lead and someone had to review things. (I don't mind it, it just feels odd for 4 small commits.)

tatiana commented:
> I'm not so sure that Harel should have this status -- he hasn't contributed directly to the project much. My understanding was his involvement was because he was a team lead, and someone had to review things. (I don't mind it, it just feels odd for 4 small commits.)

Thanks a lot, @ashb , for your thoughts on this! I addressed the feedback in fb2fcc4acb04c425cd89e53e7de6ba9a7154c61a, and I'll merge this. If we believe we were unfair in the future, we can always revisit this.
